algosdk/atomic_transaction_composer.py

```diff
@@ -165,67 +188,156 @@ def add_method_call(
+                "One of the following application creation parameters were set on a non-creation call: approvalProgram, clearProgram, numGlobalInts, numGlobalByteSlices, numLocalInts, numLocalByteSlices, extraPages"
+            )
         if not method_args:
             method_args = []
         if len(method.args) != len(method_args):
             raise error.AtomicTransactionComposerError(
                 "number of method arguments do not match the method signature"
             )
-        if not isinstance(method, abi.method.Method):
+        if not isinstance(method, abi.Method):
             raise error.AtomicTransactionComposerError(
                 "invalid Method object was passed into AtomicTransactionComposer"
             )
-        if app_id == 0:
-            raise error.AtomicTransactionComposerError(
-                "application create call not supported"
-            )
+
+        # Initialize foreign object maps
+        if not accounts:
+            accounts = []
+        if not foreign_apps:
+            foreign_apps = []
+        if not foreign_assets:
+            foreign_assets = []
 
         app_args = []
-        # For more than 14 args, including the selector, compact them into a tuple
-        additional_args = []
-        additional_types = []
+        raw_values = []
+        raw_types = []
         txn_list = []
         # First app arg must be the selector of the method
         app_args.append(method.get_selector())
         # Iterate through the method arguments and either pack a transaction
         # or encode a ABI value.
         for i, arg in enumerate(method.args):
-            if arg.type in abi.method.TRANSACTION_ARGS:
+            if abi.is_abi_transaction_type(arg.type):
                 if not isinstance(method_args[i], TransactionWithSigner):
                     raise error.AtomicTransactionComposerError(
                         "expected TransactionWithSigner as method argument, but received: {}".format(
                             method_args[i]
                         )
                     )
-                if method_args[i].txn.group and method_args[i].txn.group != 0:
-                    raise error.AtomicTransactionComposerError(
-                        "cannot add a transaction with nonzero group ID"
-                    )
                 txn_list.append(method_args[i])
-            elif len(app_args) > self.MAX_ABI_APP_ARG_LIMIT:
-                # Pack the remaining values as a tuple
-                additional_types.append(arg.type)
-                additional_args.append(method_args[i])
             else:
-                encoded_arg = arg.type.encode(method_args[i])
-                app_args.append(encoded_arg)
+                if abi.is_abi_reference_type(arg.type):
+                    current_type = abi.UintType(8)
+                    if arg.type == abi.ABIReferenceType.ACCOUNT:
+                        account_arg = str(method_args[i])
```

Comment:
As we discussed, this should "normalize" the input before examinging it, since the input could be a 32-length byte array or a base32 address string